### PR TITLE
Fix Outfitter Attribute display bug

### PR DESCRIPTION
**Pull request by tehhowch**
Refs #3147 

The fix is to not combine the `const char *` value `it.first` and `':'`. Previously, `it.first` was a `std::string`, but after 595d2e8557489ae4549e42e3eb42a5ac7eecd04d it became `const char *`. Due to use of `auto` for the iterator `it`, the pointer to char was kept, rather than converting to a string for display / use.

Summary:
 - Change `auto` to the type `pair<string, double>` when `it.first` is to be used as a string
 - `push_back` to `emplace_back`
 - Simplify display of per-second values
 - Omit per-second values if reload = 60 (i.e. per second == per shot)
 - Separate non per shot / per second values from values which should be printed as per shot / per second
    - Addition of new damage types no longer requires updating the `isContinuous` offset

### CODE REVIEW

**Comment by tehhowch** [comment]
I dropped printing the per second version when per second == per shot because then it is apparent that these values are needed each time the outfit fires, and because continuous weapons will look different than discrete weapons.

**Comment by endless-sky** [comment]
Given how few weapons fire exactly once per second, I feel like it would make sense to always show values per second, and only drop values per shot when the weapon fires continuously. (Especially because damage per second is what the player will use for comparing weapons, so it'll be really confusing to them if one weapon shows no DPS values.)

**Comment by tehhowch** [comment]
I'm tempted to try my hand at reordering the attributes, but I'll leave that to you following the inputs of #3108

**Comment by tehhowch** [comment]
Made a mockup - I have a stashed commit I can push if its OK:
```
{non-weapon attributes}

ammo
range
inaccuracy
blast radius
turret turn
anti-missile
missile strength
homing
tracking
piercing

shots / sec
{per sec damage & usage}

{per shot damage & usage}
```
Detainer:
image

Disruptor:
image

Finisher Pod:
image

Point Defense:
image

File: source/OutfitInfoDisplay.cpp [hunk 1]
```diff
@@ -171,147 +171,117 @@ void OutfitInfoDisplay::UpdateAttributes(const Outfit &outfit)
 	attributeValues.clear();
 	attributesHeight = 20;
 	
-	map<string, map<string, int>> listing;
-	for(const auto &it : outfit.Attributes())
+	for(const pair<string, double> &it : outfit.Attributes())
```
Comment (endless-sky) [inline source/OutfitInfoDisplay.cpp]: When you do this, what's actually happening is that you're constructing a copy of each attributes node (via pair's templated constructor) and then getting a const reference to that temporary (which extends its lifetime) and then using that reference. That's a lot of behind-the-scenes stuff to hide behind a for loop. I think it'd make more sense to just iterate through the actual items in the attributes set (which are pair<const char *, double>s) and then typecast the name to a string if needed for concatenating strings.

Comment (tehhowch) [reply source/OutfitInfoDisplay.cpp]: That was how I had it originally, and then when I changed it to test if there would have been a warning when the type of `it` changed from <string, double> to <const char *, double> I found that no warning was generated (and no display bug from adding `':'` to the char * was present either), so I stuck with it.
I'll swap back

**Comment by tehhowch** [comment]
@endless-sky addressed.
Thoughts on if I should push the reordering as well?

**Comment by endless-sky** [comment]
I think I'm going to experiment with the idea of putting the values in a table, which would probably conflict with any reordering (or at least very quickly make it obsolete).

**Pull request merged**